index.d.ts

```diff
@@ -0,0 +1,104 @@
+// TypeScript Version: 3.7
+
+import {Test} from 'unist-util-is'
+import * as unist from 'unist'
+
+export {}
+
+type Null = undefined | null
+
+declare namespace messageControl {
+  export {}
+  /** a comment marker */
+  export interface Marker<N extends unist.Node> {
+    /** name of marker */
+    name: string
+    /** value after name */
+    attributes: string
+    /** parsed attributes */
+    parameters: Record<string, unknown>
+    /** reference to given node */
+    node: N
+  }
+
+  /** parse a possible comment marker node to a Marker */
+  export interface MarkerFn<N extends unist.Node> {
+    (node: N): Marker<N> | Null
```

Comment:
`MarkerFn` should return `null` not `null` or `undefined`.

Reply:
Happy to change this but it seems to intersect with the discussion on how `null` is / should be treated in the unified ecosystem